### PR TITLE
Tech Task: Update readme with standard deploy process

### PR DESCRIPTION
**Pull request by giladshanan**
# Release Notes

- Adds `latest_from_open` script
- Adds stage url
- Updates deployment steps

### CODE REVIEW

File: doc/HOWTO_forks.md [hunk 1]
```diff
@@ -23,7 +23,13 @@ git fetch upstream
 git merge upstream/master
 ```
 
-Push up your new branch, and run `bin/merge_describer` which generates a helpful description you can copy/paste into your PR; it contains the commit messages and links to the merged branches since the last merge from open.
+Run `bin/merge_describer` which generates a helpful description you can copy/paste into your PR; it contains the commit messages and links to the merged branches since the last merge from open.
+
+Push up your new branch, and paste the output from `bin/merge_describer` into the PR description.  For clarity it's best to sort commits into 3 groups: features, fixes, and tech tasks.
+
+After adding the `upstream` remote, you can use the `bin/latest_from_open` script to automate creating a new `latest_from_open_YYYYMMDD` branch.
```
Comment (giladshanan) [inline doc/HOWTO_forks.md:30]:
```suggestion
After adding the `upstream` remote, you can use the `bin/latest_from_open` script to create a new `latest_from_open_YYYYMMDD` branch.
```

File: bin/latest_from_open [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/sh
```
Comment (rdunlop) [inline bin/latest_from_open:1]: This file is not executable. Please make it executable?

**Review by rdunlop — CHANGES_REQUESTED** [review]

File: doc/HOWTO_forks.md [hunk 3]
```diff
@@ -23,7 +23,13 @@ git fetch upstream
 git merge upstream/master
 ```
 
-Push up your new branch, and run `bin/merge_describer` which generates a helpful description you can copy/paste into your PR; it contains the commit messages and links to the merged branches since the last merge from open.
+Run `bin/merge_describer` which generates a helpful description you can copy/paste into your PR; it contains the commit messages and links to the merged branches since the last merge from open.
+
+Push up your new branch, and paste the output from `bin/merge_describer` into the PR description.  For clarity it's best to sort commits into 3 groups: features, fixes, and tech tasks.
+
+After adding the `upstream` remote, you can use the `bin/latest_from_open` script to create a new `latest_from_open_YYYYMMDD` branch.
```
Comment (giladshanan) [inline doc/HOWTO_forks.md:30]:
```suggestion
After adding the `upstream` remote, you can use the `bin/latest_from_open` script to create a new `latest_from_open_MMDDYYYY` branch.
```

**Comment by giladshanan** [comment]
@rdunlop any further feedback?

File: doc/HOWTO_forks.md [hunk 4]
```diff
@@ -43,19 +49,47 @@ git cherry-pick XXXXX
...
+General process:
+1. Open a `latest_from_open_MMDDYYYY` branch
+1. Merge the PR (not squash) once it has been approved and CI is green
+1. For a production release, create a Github release:
+  - Check out the commit you want to release and tag it: `git tag v2021-02-18`
+  - Use `bin/merge_describer v2021-01-28` to list all commits since the previous release tag
+  - Organize the output into features, fixes and tech tasks
+  - Create a new release in Github using this list of commits for the description
+  - Specify the new tag (`v2021-02-18`) for the release tag version, target, and release title
+1. Deploy the new code using capistrano or helm/CircelCI (see below)
```
Comment (rdunlop) [inline doc/HOWTO_forks.md:62]: typo CircleCI

**Review by rdunlop — APPROVED** [review]

**Review by KaraAJC — APPROVED** [review]

File: doc/HOWTO_related_repos.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+NUcore is usually deployed from a school-specific repo containing customized branding
+and functionality. There are situations where you may need to bring changes
+from the `nucore-open` repo into a school-specific repo, or the other way around.
+In order to move commits between the repos you'll need to set up remote
+repositories on each side.
+
+## Bringing changes from `nucore-open` into a school-specific repo
+
+You can bring over all the new changes with a `latest_from_open_MMDDYYYY` branch,
+or just a few commmits with a cherry-picked branch.
```
Comment (wangminghao) [inline doc/HOWTO_related_repos.md:12]: You can either bring over....
or just a few commits...

File: doc/HOWTO_related_repos.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+### General process
+
+1. Start in the directory of the school repo (for example, `nucore-nu`)
+1. Open a new branch, either`latest_from_open_MMDDYYYY` or cherry-picked [(see below)](#creating-a-latest_from_open_mmddyyyy-branch)
+1. Merge the PR (not squash) once it has been approved and CI is green [(see below)](#squash-vs-merge)
+1. For a production release, create a Github release [(see below)](#creating-a-github-release)
+1. Deploy the new code using capistrano or helm/CircleCI [(see below)](#deploy)
+1. Run any relevant rake tasks
```
Comment (wangminghao) [inline doc/HOWTO_related_repos.md:21]: (e.g. new env variables, one-off data fix...)

File: doc/HOWTO_related_repos.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+## Bringing changes from `nucore-open` into a school-specific repo
+
+You can bring over all the new changes with a `latest_from_open_MMDDYYYY` branch,
+or just a few commmits with a cherry-picked branch.
+
+### General process
+
+1. Start in the directory of the school repo (for example, `nucore-nu`)
+1. Open a new branch, either`latest_from_open_MMDDYYYY` or cherry-picked [(see below)](#creating-a-latest_from_open_mmddyyyy-branch)
```
Comment (wangminghao) [inline doc/HOWTO_related_repos.md:17]: please add a link to the cherry-pick section below

File: doc/HOWTO_related_repos.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+git co -b upgrade_ruby_2_6_6
+git fetch upstream
+git cherry-pick XXXXX
+```
+
+## Creating a `latest_from_open_MMDDYYYY` branch
+
+_While in your school-specific repo's directory_
+
+1. Add a new remote called `upstream ` that points to the `nucore-open` repo
```
Comment (wangminghao) [inline doc/HOWTO_related_repos.md:38]: extra space

File: doc/HOWTO_related_repos.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+### Notes
+
+You can use the [`bin/latest_from_open`](bin/latest_from_open) script to create a new `latest_from_open_MMDDYYYY` branch instead of using the steps above.  Add the `upstream` remote, then run `bin/latest_from_open 04012021`.
+
+When merging in changes from the open source repo, differences in `scehma.rb` and `Gemfile.lock` tend to be the trickiest to resolve.  Run `bundle` and `rails db:migrate` locally to ensure any conflicts have been resolved as expected.
+
+## Squash vs Merge
+
+On github, squash commits include a parenthesized branch number like `(#45)` in the commit message by default, except when the PR only includes 1 commit. Merge commits do not include the branch number.
+The `bin/merge_describer` script only describes commits which include the parenthesized branch number,
```
Comment (wangminghao) [inline doc/HOWTO_related_repos.md:54]: Make this the first point - it feels that important

File: doc/HOWTO_related_repos.md [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+
+# Bringing changes from your school-specific repo into `nucore-open`
+
+_While in the `nucore-open` directory_
+
+For example, to bring changes from `nucore-nu` into `nucore-open`:
+
+`git remote add nu git@github.com:tablexi/nucore-nu.git`
+
+The best way to bring in the changes from NU is by cherry-picking the individual
```
Comment (wangminghao) [inline doc/HOWTO_related_repos.md:75]: long discussion....

File: doc/HOWTO_related_repos.md [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+commits. For this reason, it's best to develop on `nucore-open` to start with, or to squash
+the commits in your branch so you only need to bring one commit.
+
+```
+git fetch nu
+git cherry-pick XXXXX
+```
+
+# Deploy
+
```
Comment (wangminghao) [inline doc/HOWTO_related_repos.md:85]: If there is a preferred deployment method? "Method may vary depending on which school"

**Review by wangminghao — APPROVED** [review]

**Pull request merged**